### ISSUE TITLE
[crash of piepmatz]

### ISSUE THREAD

**Issue by Nokius** [Wunderfitz/harbour-piepmatz#6]
@Wunderfitz it happened this morning again the random crash.

My observing are.

* refresh of the timeline (via Cover)
* scrolling down the new tweets 

--> crash

**Comment by Wunderfitz** [comment]
I also had this during testing quite a few times, but it wasn't really reproducible. Well, let's see if I can find the culprit...

**Comment by TMavica** [comment]
it happened for me too, sometimes if long time no refresh the cover, it crashed

**Comment by Nokius** [comment]
I see patterns, next week I will run piepmatz via cli and check the output, atm it looks like an memory leak or an issue with qml which hangs piepmatz.

@TMavica good luck with the A350 ;)

**Comment by Nokius** [comment]
This week I had it several times but I missed to log piepmatz sorry

**Comment by Wunderfitz** [comment]
I've only had it once while my device was connected to Qt Creator. Not that much to see:
`bash: line 1: 20451 Segmentation fault      DISPLAY=':0.0' QT_NO_JOURNALD_LOG='1' /usr/bin/harbour-piepmatz`
Obviously, I need to learn how to trace segmentation faults...

**Comment by Nokius** [comment]
it happes to me once a day, I guess it's related to image/gif caching

**Comment by Nokius** [comment]
So fare it didn't happend with the new version looks like you fixed the issue :)

**Comment by Wunderfitz** [comment]
I wouldn't be that sure. I didn't do anything on this side as I still don't have a clue what the root cause is. Moreover, it happened for me on my FP2 at least once with the new version. However, it never seems to happen when I connected my device to the SailfishOS IDE to trace things... :(

At least, Piepmatz should now remember the list position so you can continue where you were after a crash.

**Comment by Wunderfitz** [comment]
OK, I tested a fix over several days and it seems that the crashes are really gone. Was pretty hard to detect as there were no proper stack traces. However, I had a suspicion about the way how I handle videos. Obviously, the OS wasn't very happy about that. I changed the video handling and also improved the playback UI a bit and that was it...

Have fun with 0.4 - hopefully without the crashes/freezes, or at least with far less of them!
